lib/dal/src/pkg/import.rs

```diff
@@ -2543,14 +2453,38 @@ async fn import_schema_variant(
                         ),
                     }
                 }
-                None => (
-                    // FIXME(nick): move category, color, and all metadata to variant or somewhere
-                    // else. It should not be on schema.
-                    SchemaVariant::new(ctx, schema.id(), variant_spec.name(), category)
+                None => {
+                    let spec = schema_spec.to_spec().await?;
+                    let metadata = SchemaVariantJson::metadata_from_spec(spec)?;
+
+                    let mut asset_func_id: Option<FuncId> = None;
+                    if let Some(variant_spec_data) = variant_spec.data() {
+                        let func_unique_id = variant_spec_data.func_unique_id().to_owned();
+                        if let Thing::Func(asset_func) = thing_map
+                            .get(change_set_id, &func_unique_id)
+                            .ok_or(PkgError::MissingFuncUniqueId(func_unique_id.to_string()))?
+                        {
+                            asset_func_id = Some(asset_func.id)
+                        }
+                    }
+                    (
+                        SchemaVariant::new(
+                            ctx,
+                            schema.id(),
+                            variant_spec.name(),
+                            metadata.menu_name,
+                            metadata.category,
+                            metadata.color,
+                            metadata.component_type,
+                            metadata.link,
+                            metadata.description,
+                            asset_func_id,
+                        )
                         .await?
                         .0,
```

Comment:
nit: feel free to ignore. It's unclear what the tuple returns. I'd put it in a variable.